bundle/config/mutator/apply_transforms.go

```diff
@@ -0,0 +1,179 @@
+package mutator
+
+import (
+	"context"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/textutil"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/databricks/databricks-sdk-go/service/ml"
+)
+
+type applyTransforms struct{}
+
+// Apply all transforms, e.g. the prefix transform that
+// adds a prefix to all names of all resources.
+func ApplyTransforms() *applyTransforms {
+	return &applyTransforms{}
+}
```

Comment:
This approach actually seemed easier for maintainers. Adding a new resource is pretty hard right now since you need to contribute to so many places. With the approach here you can see at a glance what was done for other resource types and mimic what was done there.

Reply:
Sure, we can stick with this for now but I can see it growing unmanageable. There is a `ConfigResource` interface that all resources need to implement that could be extended for each one of these options. Then the code won't even compile without implementing the expected set of functions.

Can keep as is for this PR.